### ISSUE TITLE
Same song repeated over and over again

### ISSUE THREAD

**Issue by tsdh** [memen45/SubMusic#94]
Hi, thanks a lot for SubMusic.  I use it daily and love it.  🫶

I've notices one strange issue.  I have 4 playlists with about 20 songs each on the watch.  I always use shuffle and "repeat all".  Once or twice a week, I get into the situation where SubMusic will repeat the same song over and over again (the settings still saying "repeat all" and not "repeat 1").  Even when pressing the "skip to next song" button, it will just restart the current song.  In order to get back into normal operation, I have to "Play All" again.

I have the assumption that this might happen after all my 80 downloaded songs have been played.  I would expect that (due to "repeat all") it would start again with randomly choosing the next song out of the 80 songs, then the next out of the 79 remaining, and so forth.

**Comment by memen45** [comment]
Thanks for your feedback and kind words! The issue you are describing is interesting. Unfortunately, the specific details for repeat 1 and repeat all are not documented by Garmin at all, so the implementation was only tested on the simulator, and not on all devices. Did you happen to test with a playlist with e.g. 3 songs? Or does it only happen when choosing play all option?

**Comment by tsdh** [comment]
Sorry for not responding for a long time.  The "problem" is that I started listening to podcasts (using the runcasts app) in addition to music.  And since I'm now regularly switching between these two apps, it happens much less often or basically never that I listen through one complete playlist or even all playlists.  Obviously, I could test by deleting my playlist and downloading just a short playlist but since I have general wifi download issues with the watch, I'm happy for the content that already made it there. ;-)

So basically my change in habits solved that issue for me.  But maybe someone else can contribute by testing with a shorter playlist.
